### PR TITLE
Fix email notification error with StripeLink button code 

### LINKED ISSUE #4675: StripeLink: fix email notification error
### Describe the bug

On the checkout page, the email empty field error notification also includes HTML tags. This was introduced when the Link button was implemented.

### To Reproduce

1. Go to checkout page
2. Fill all required fields, except for email
3. Try to place the order

### Actual behavior

The actual error notification: `Billing Email address <button class="wcpay-stripelink-modal-trigger"></button> is a required field.`
### Screenshots

Screenshot 2022-09-01 at 10 36 21

### Expected behavior

The error notification that should be displayed: `Billing Email address is a required field.`
### Desktop (please complete the following information):

* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

### Smartphone (please complete the following information):

* Device: [e.g. iPhone6]
* OS: [e.g. iOS8.1]
* Browser [e.g. stock browser, safari]
* Version [e.g. 22]

### PR DESCRIPTION
**Pull request by nagpai**
Fixes #4675

#### Changes proposed in this Pull Request

Prevent button HTML code from appearing in the error notification, when the email address is blank. 

 #### Testing instructions

1. Make sure you have StripeLink feature enabled within `Payments > Settings`, under Express checkouts.
image

2. Create a new order from front-end, and browse to the checkout page
3. Fill in all required details, except email address and click the `Place order` button
4. You should see an error message `Billling Email address is a required field.` , without any unwanted HTML code within it. 

##### Before 
image

##### After
image - after

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply) - Tested with a resized browser window. Link button appears fine in its intended place. 

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by nagpai** [comment]
Props to @dpaun1985  for guiding me with the correct hook to fix this issue 🙌

**Review by mgascam — APPROVED** [review]
LGTM! 
✅ Followed testing instructions and works as expected

File: changelog/fix-4675-stripelink-notification [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Remove button HTML code seen in error notification, when email address is kept blank
```
Comment (kalessil) [inline changelog/fix-4675-stripelink-notification]: `Correct empty email error when StripeLink is active on checkout page.`

**Review by kalessil — APPROVED** [review]
Per-approving. Please check my comment about the change log entry.

**Review by dpaun1985 — APPROVED** [review]
LGTM. 
✅ Tested on chrome and safari and works as expected.

File: changelog/fix-4675-stripelink-notification (hunk 1 above)
Comment (nagpai) [reply changelog/fix-4675-stripelink-notification]: Thank you! I have updated the changelog entry.

**Pull request merged**